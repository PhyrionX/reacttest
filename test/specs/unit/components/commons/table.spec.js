import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import Table from '../../../../../src/components/commons/table';

console.error = (message) => {
  if (/(Failed prop type)/.test(message)) {
    throw new Error(message);
  }
};

describe('<Table />', () => {
  it('should not mount the component', () => {
    try {
      render(<Table />);
    } catch (e) {
      expect(e instanceof Error).toEqual(true);
    }
  })

  it('should mount the component', () => {
    const { asFragment } = render(<Table columns={ [
        {
          label: 'test1',
          key: 'test1'
        },
        {
          label: 'test2',
          key: 'test2',
          type: 'date'
        }
      ] }
      data={[
        { test1: 'a', test2: '2020-03-28T17:28:59.428Z' },
        { test1: 'b', test2: '2020-03-28T17:28:59.428Z' }
      ]} />);

    expect(asFragment()).toMatchSnapshot();
  })

  it('should not mount the component and row click', () => {
    const mockClick = jest.fn();
    const { container } = render(<Table columns={ [
        {
          label: 'test1',
          key: 'test1'
        },
        {
          label: 'test2',
          key: 'test2',
          type: 'date'
        }
      ] }
      data={[
        { test1: 'a', test2: '2020-03-28T17:28:59.428Z' },
        { test1: 'b', test2: '2020-03-28T17:28:59.428Z' }
      ]}
      onRowClick={ mockClick } />);
    const td = container.querySelector('.tfg-table__td--clickable');
    
    fireEvent.click(td);
    
    expect(mockClick).toHaveBeenCalled();
  })
})