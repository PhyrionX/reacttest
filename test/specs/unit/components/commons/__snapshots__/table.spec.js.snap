// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should mount the component 1`] = `
<DocumentFragment>
  <table
    class="table tfg-table"
  >
    <thead>
      <tr>
        <th>
          test1
        </th>
        <th>
          test2
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class=""
        >
          a
        </td>
        <td
          class=""
        >
          March 28th 2020, 6:28:59 pm
        </td>
      </tr>
      <tr>
        <td
          class=""
        >
          b
        </td>
        <td
          class=""
        >
          March 28th 2020, 6:28:59 pm
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
